src/components/scene/xr-mode-ui.js

```diff
@@ -150,6 +150,7 @@ module.exports.Component = registerComponent('xr-mode-ui', {
     } else {
       if (!utils.device.checkVRSupport()) { this.enterVREl.classList.add('fullscreen'); }
       this.enterVREl.classList.remove(HIDDEN_CLASS);
+      sceneEl.enterVR(false, true);
```

Comment:
We should probably handle this a bit differently. Give control to the dev and decide if the page is auto-entering VR. Maybe a property in this component `enterImmersiveModeAtPageLoad` a tad long but descriptive. open to a shorter alternative. We can then probably handled the sceneEl.enterVR here https://github.com/aframevr/aframe/pull/5410/files#diff-8eae4e9d86bc7d08e7db133212411b40a655722a0d8c42c1c99c45fccef72f74L124 before deciding to display the UI